apps/coscrad-frontend/src/components/nav-menu/nav-menu-presenter.tsx

```diff
@@ -0,0 +1,58 @@
+import MenuRoundedIcon from '@mui/icons-material/MenuRounded';
+import { Box, IconButton, Menu } from '@mui/material';
+import { useState } from 'react';
+import AuthenticationButton from '../authentication-button/authentication-button';
+import { NavItemInfo } from './nav-menu-container';
+import { NavMenuItem } from './nav-menu-item';
+
+interface NavMenuPresenterProps {
+    navItemInfos: NavItemInfo[];
+}
+
+export const NavMenuPresenter = ({ navItemInfos }: NavMenuPresenterProps): JSX.Element => {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const open = Boolean(anchorEl);
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleClose = () => {
```

Comment:
Really a shame that this is necessary, but you handled it pretty cleanly.

Reply:
Not sure what do do about it for now, but maybe wrap the `<Menu>` component in another one that has the menu state in it.  Also, I realize that it's a bit of a funny implementation that I grabbed.  It checks if the anchor is null (as you pointed out), but in reality you could separate open and close from whether or not the anchor is defined or not.  Not sure if there's another pattern we could use here.

Reply:
I'll leave it as is for now.

Reply:
It's ok for now. Let's worry about it next pass if it causes any trouble \ confusion.